inc/agent.class.php

```diff
@@ -545,6 +563,34 @@ public function prepareInputForUpdate($input) {
          }
       }
 
+      if ($this->canUpdateItem() && PluginFlyvemdmCommon::checkAgentResponse($input)) {
```

Comment:
You should not call can*Item() in prepareInputFor* because there is a separation between rights checks and  preparation of $input .

calling can*Item() is already done by the UI or the API then your call slows down the code. Also if the code internally needs to ad or update an object as a consequence of an other process, it is possible that checking the rights check will fail when we need to succeed.

These reasons lead me to submit a PR on GLPI as I told you on telegram.

You should avoid as much as possible rights checks here.